File: ios-native/NeobankNative/Sources/Features/Wallet/WalletRepository.swift
import Foundation

struct WalletRepository: Sendable {
    let client: APIClient

    func balance(currency: String = "USD") async throws -> WalletBalance {
        try await client.send("/v1/wallet", query: ["currency": currency])
    }

    func transactions(cursor: String? = nil, limit: Int = 20) async throws -> WalletTransactionPage {
        try await client.send(
            "/v1/wallet/transactions",
            query: ["limit": String(limit), "cursor": cursor]
        )
    }
}

File: android-native/app/src/main/kotlin/dev/horobets/neobanknative/features/wallet/WalletRepository.kt
package dev.horobets.neobanknative.features.wallet

import dev.horobets.neobanknative.core.networking.ApiClient

class WalletRepository(private val client: ApiClient) {
    suspend fun balance(currency: String = "USD"): WalletBalance =
        client.send("/v1/wallet", query = mapOf("currency" to currency))

    suspend fun transactions(cursor: String? = null, limit: Int = 20): WalletTransactionPage = client.send(
        "/v1/wallet/transactions",
        query = mapOf("limit" to limit.toString(), "cursor" to cursor),
    )
}
